hot-reload-analysis/src/main/kotlin/org/jetbrains/compose/reload/analysis/utils.kt

```diff
@@ -36,3 +36,11 @@ internal fun Checksum.updateString(value: String) {
     updateInt(value.length)
     update(value.encodeToByteArray())
 }
+
+internal fun <K, V> HashMap<K, V>.toImmutableHashMap(): Map<K, V> = when (size) {
+    0 -> emptyMap()
+    1 -> mapOf(entries.first().toPair())
```

Comment:
Maybe using `singletonMap` directly here would make it clearer? This way, a reader needs to know that mapOf with this single pair is optimizing by creating a Singleton Map